src/checkbox/checkbox.tsx

```diff
@@ -49,13 +49,19 @@ export const Checkbox = ({
             className={className}
             data-testid="checkbox"
             $displaySize={displaySize}
+            role="checkbox"
+            aria-checked={selected}
+            aria-labelledby="checkbox-input"
+            tabIndex={disabled ? -1 : 0}
         >
             <Input
-                type="checkbox"
+                id="checkbox-input"
                 data-testid="checkbox-input"
+                aria-hidden="true"
+                type="checkbox"
+                tabIndex={-1}
                 onChange={handleOnCheck}
```

Comment:
you will need special keypress handling since the tab focus is now no longer on the actual input. I am unable to select the input using the keyboard.